tests/test_upstream_hub.py

```diff
@@ -77,7 +85,9 @@ def test_push_dataset_dict_to_hub_no_token(self):
 
             # Ensure that there is a single file on the repository that has the correct name
             files = sorted(self._api.list_repo_files(ds_name, repo_type="dataset"))
-            self.assertListEqual(files, [".gitattributes", "data/train-00000-of-00001.parquet", "dataset_infos.json"])
+            self.assertFilesMatch(
+                files, [".gitattributes", "data/train-00000-of-00001-<FINGERPRINT>.parquet", "dataset_infos.json"]
+            )
```

Comment:
(nit)
It's maybe simpler to match it using a wildcard `*` (maybe with `fnmatch`) instead of implementing a custom logic that reads until the last dash ? And this way you don't have to have a different behavior for files inside `data/` and the others.
```suggestion
            self.assertFilesMatch(
                files, [".gitattributes", "data/train-00000-of-00001-*.parquet", "dataset_infos.json"]
            )
```